flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/snapshot/RocksIncrementalSnapshotStrategy.java

```diff
@@ -182,7 +186,14 @@ private SnapshotDirectory prepareLocalSnapshotDirectory(long checkpointId) throw
 		if (localRecoveryConfig.isLocalRecoveryEnabled()) {
 			// create a "permanent" snapshot directory for local recovery.
 			LocalRecoveryDirectoryProvider directoryProvider = localRecoveryConfig.getLocalStateDirectoryProvider();
-			File directory = directoryProvider.subtaskSpecificCheckpointDirectory(checkpointId);
+			String operatorId = OperatorSubtaskDescriptionText.getOperatorIdByOperatorIdentifier(operatorIdentifier);
+			if (operatorId == null) {
+				operatorId = operatorIdentifier;
+				LOG.warn("Can not extract operatorId from {}, the expected format is {}.",
+					operatorIdentifier,
+					EXTRACT_OPERATOR_PATTERN.toString());
+			}
+			File directory = directoryProvider.subtaskSpecificCheckpointDirectory(checkpointId, operatorId);
 
 			if (directory.exists()) {
```

Comment:
As commented on the JIRA already, instead of applying all the changes in the PR, wouln't it be sufficient to remove this deletion an existing root directory of the checkpoint? This looks like the actual cause of the problem to me. What do you think? We could check and delete an existing subdirectory for the particular operator instead.